lib/iris/options.py

```diff
@@ -0,0 +1,107 @@
+# (C) British Crown Copyright 2017, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Control runtime options of Iris.
+
+"""
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+import six
+
+import contextlib
+import warnings
+
+
+class NetCDF(object):
```

Comment:
Will we though? It is interesting that dask's `set_options` is **not** thread-safe. I left this like this for precisely this reason -- so that we'd get some commenting on it.

In fact I'd go so far as to argue that we _don't_ want options thread-safe (!) -- my reason being that if I set an option I'd like it to be available in **all** threads, not just the calling thread. Of course, this may indicate some mixed-up thinking about how threads work, so I'm happy to stand corrected about this one.